src/main/java/mate/academy/rickandmorty/service/RickAndMortyClient.java

```diff
@@ -0,0 +1,37 @@
+package mate.academy.rickandmorty.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.CharacterResponseDto;
+import mate.academy.rickandmorty.dto.ListCharacterDto;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class RickAndMortyClient {
+    private static final String CHARACTER_URL = "https://rickandmortyapi.com/api/character";
+    private final ObjectMapper objectMapper;
+
+    public List<CharacterResponseDto> getCharacters() {
+        HttpClient httpClient = HttpClient.newHttpClient();
+        HttpRequest request = HttpRequest.newBuilder()
+                .GET()
+                .uri(URI.create(CHARACTER_URL))
+                .build();
+        try {
+            HttpResponse<String> response = httpClient
+                    .send(request, HttpResponse.BodyHandlers.ofString());
+            ListCharacterDto listCharacterDto = objectMapper
+                    .readValue(response.body(), ListCharacterDto.class);
+            return listCharacterDto.getResults();
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
Add method description here.
```suggestion
            throw new RuntimeException(e);
```